haystack/dataclasses/chat_message.py

```diff
@@ -297,10 +332,22 @@ def to_dict(self) -> Dict[str, Any]:
         :returns:
             Serialized version of the object.
         """
+
+        # We don't want to show the warning if ChatMessage.to_dict is used in pipeline serialization
+        used_in_pipeline_serialization = any(
+            frame.function == "component_to_dict" and "serialization.py" in frame.filename for frame in inspect.stack()
+        )
+        if not used_in_pipeline_serialization:
+            logger.warning(
+                "Starting from Haystack 2.12.0, ChatMessage.to_dict returns a dictionary with keys 'role', "
+                "'meta', 'name', and 'content' instead of '_role', '_meta', '_name', and '_content'. "
+                "If your code consumes this dictionary, please update it to use the new format."
+            )
```

Comment:
@sjrl @davidsbatista this is the only way I found to show the warning only if this method is not called in pipeline serialization.
I don't like it, but at the same time, I don't want to show this warning on each Pipeline serialization involving a `ChatPromptBuilder`.
WDYT?

Reply:
I'm OK with it - this will still be only temporary and in-between releases, right?

Reply:
I would make it visible in 2.12.0 (and maybe 2.13.0) and remove it afterward.

Reply:
from my side this approach is OK - nothing against it - and thanks for investigating it 👍🏽

Reply:
also good from my end